Source/DafnyCore/Resolver.cs

```diff
@@ -15074,7 +15074,8 @@ public ResolutionContext(ICodeContext codeContext, bool isTwoState)
         ResolveExpression(e.Index, resolutionContext);
         ResolveExpression(e.Value, resolutionContext);
         AddXConstraint(expr.tok, "SeqUpdatable", e.Seq.Type, e.Index, e.Value, "update requires a sequence, map, or multiset (got {0})");
-        expr.Type = e.Seq.Type;
+        var ty = PartiallyResolveTypeForMemberSelection(expr.tok, e.Seq.Type);
+        expr.Type = ty;
```

Comment:
This does not look right to me — there's no member selection here.  Instead, shouldn't we just revert to the base type? (`e.Seq.Type` with the constraints removed?)